### ISSUE TITLE
Test failure in test_function_factory.py

### ISSUE THREAD

**Issue by Balandat** [cornellius-gp/gpytorch#94]
The offending line is https://github.com/cornellius-gp/gpytorch/blob/master/test/util/test_function_factory.py#L243

The last `.dot` operation errors with 
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-28-1e6c865d4b6a> in <module>()
----> 1 t.test_inv_quad_log_det_many_vectors()

<ipython-input-25-045795d3c376> in test_inv_quad_log_det_many_vectors(self)
     24     def test_inv_quad_log_det_many_vectors(self):
     25         # Forward pass
---> 26         actual_inv_quad = self.mat_var_clone.inverse().matmul(self.vecs_var_clone).dot(self.vecs_var_clone)
     27         with gpytorch.settings.num_trace_samples(1000):
     28             nlv = NonLazyVariable(self.mat_var)

RuntimeError: Expected argument self to have 1 dimension, but has 2
```

I'm not entirely sure what this code does, so I'll leave this for @gpleiss to fix.

Not sure if that happens on stable pytorch, but it definitely happens on pytorch master. Given that docs back to 0.3.0 state that .dot does not broadcast, I'm assuming this is unrelated though.

**Comment by gpleiss** [comment]
Looking into this now. I'm sure we can add some branching code that should satisfy pytorch 0.3 and pytorch master

**Comment by gpleiss** [comment]
Oh whoops I just realized #94 is addressing this issue :grin:

**Comment by gpleiss** [comment]
I'm going to try to add PyTorch master to our list of Travis checks. Hopefully then we can avoid issues like this in the future (or at least until 0.4 becomes an official release).

**Comment by Balandat** [comment]
Great, I was about to suggest doing that.

For the record: This is solved in #97, not #95
